src/v/config/configuration.cc

```diff
@@ -470,6 +470,14 @@ configuration::configuration()
       "write with the given producer id.",
       {.visibility = visibility::user},
       10080min)
+  , max_concurrent_producer_ids(
+      *this,
+      "max_concurrent_producer_ids",
+      "Max cache size for pids which rm_stm stores inside internal state. In "
+      "overflow rm_stm will delete old pids and clear their status",
+      {.needs_restart = needs_restart::no, .visibility = visibility::tunable},
+      std::numeric_limits<uint64_t>::max(),
```

Comment:
seems like we should choose something more modest than int64_t::max for the max size of the cache here.

also, should this be named something that relates to it controlling a cache size rather than using the word "concurrent" which feels like it controls something related to CPU resource?

Reply:
We need to do follow up work to define the right limit, this PR is about a a mechanism to control it. What do you suggest @dotnwat `max_parallel_producer_ids`?

Reply:
I don't know enough about this work to know of a good limit. But anytime I see a cache with an unlimited size it raises concerns.